spec/SM/StateMachine/StateMachineSpec.php

```diff
@@ -204,4 +204,24 @@ function it_returns_possible_transitions($object, $callbackFactory, CallbackInte
 
         $this->getPossibleTransitions()->shouldReturn(array('create', 'confirm'));
     }
+
+    function it_can_accept_specific_from_to_transition_mappings($object, $dispatcher, $callbackFactory, CallbackInterface $guard)
+    {
+        $transition = $this->config['transitions']['create'];
+
+        $this->config['transitions']['create'] = array(
+            $transition
```

Comment:
Looks like you only test that an array of 1 transition works the same as before.
Could you add a test for the case where a transition name contains multiple transitions defined on different `from` states? So that we're sure that we call the right transition, depending on the current state.